### ISSUE TITLE
`test_workspace_config` test fails

### ISSUE THREAD

**Issue by Kladki** [obi1kenobi/cargo-semver-checks#1509]
### Steps to reproduce the bug with the above code

Version being used: 0.45.0
The test setup can be seen [here](https://gitlab.alpinelinux.org/alpine/aports/-/merge_requests/91293/diffs#578989345b01b64370eece0739a827005eaacff7_0_31) (Just running `./scripts/regenerate_test_rustdocs.sh`, followed by `cargo test --frozen`)

### Actual Behaviour

When trying to package `cargo-semver-checks` for Alpine Linux, the aforementioned test failed on all architectures, with the following stdout:
```
running 7 tests
test test_workspace_config ... FAILED
test test_workspace_key_both_missing ... ok
test test_only_read_config_from_new_manifest ... ok
test test_workspace_key_metadata_true ... ok
test test_workspace_key_cargo_true ... ok
test test_only_warnings ... ok
test test_allow_skipped ... ok
failures:
---- test_workspace_config stdout ----
thread 'test_workspace_config' (28677) panicked at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/core/src/ops/function.rs:250:5:
Unexpected stderr, failed var.is_match(FAIL(.*)minor(.*)function_missing)
├── var: error: failed to retrieve local crate data
│   
│   Caused by:
│       0: possibly due to errors: [
│            failed to parse old/Cargo.toml: no `package` table,
│          ]
│       1: package `pkg` not found in old
└── var as str: error: failed to retrieve local crate data
    
    Caused by:
        0: possibly due to errors: [
             failed to parse old/Cargo.toml: no `package` table,
           ]
        1: package `pkg` not found in old
command=`cd "test_crates/manifest_tests/workspace_overrides" && "/builds/Matthias/aports/testing/cargo-semver-checks/src/cargo-semver-checks-0.45.0/target/debug/cargo-semver-checks" "semver-checks" "--baseline-root" "old" "--manifest-path" "new/pkg" "-v"`
code=1
stdout=""
stderr=```
error: failed to retrieve local crate data
Caused by:
    0: possibly due to errors: [
         failed to parse old/Cargo.toml: no `package` table,
       ]
    1: package `pkg` not found in old
```
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
failures:
    test_workspace_config
test result: FAILED. 6 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.04s
error: test failed, to rerun pass `--test lint_config`
```

### Expected Behaviour

For the test to pass.

### Generated System Information

#### Software version

cargo-semver-checks 0.45.0

#### Operating system

- OS: Linux (Alpine Linux 3.21.0_alpha20240923)
- Kernel: 6.6.63-0-lts

#### Command-line

```bash
/builds/Matthias/aports/testing/cargo-semver-checks/src/cargo-semver-checks-0.45.0/target/release/cargo-semver-checks semver-checks --bugreport 
```

#### cargo version

```
> cargo -V 
cargo 1.91.1 (ea2d97820 2025-10-10) (Alpine Linux Rust 1.91.1-r0)
```

#### Compile time information

- Profile: release
- Target triple: x86_64-alpine-linux-musl
- Family: unix
- OS: linux
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-alpine-linux-musl

### Build Configuration

[build]
jobs = 48
rustflags = ["-Clink-arg=-Wl,-z,pack-relative-relocs"]

**Comment by obi1kenobi** [comment]
Thanks for reporting this! It's quite a strange issue: I'm not able to reproduce it using Rust 1.91.1 on my Linux machine.

Let me try to suggest a few things you could try on your end to help us narrow down on what's causing this.

### Try removing your build configuration settings

I _doubt_ this is the cause, but it'll be very good to rule it out quickly. Build configuration has caused issues in the past, which is why it's in the issue reporting form now :)

### Try deleting the `test_crates` directory, then checking it out from git again

Sometimes stray files in the test crates directory can change test outcomes. For example, a `.cargo/config.toml` file somewhere in that directory (or above it) could definitely cause some havoc.

### Try applying this patch and see if the test outcome changes

```
diff --git a/tests/lint_config.rs b/tests/lint_config.rs
index d2684a7..96ee3e3 100644
--- a/tests/lint_config.rs
+++ b/tests/lint_config.rs
@@ -23,7 +23,7 @@ fn command_for_workspace(workspace: &str, package: &str) -> Command {
         .args([
             "semver-checks",
             "--baseline-root",
-            "old",
+            &format!("old/{package}"),
             "--manifest-path",
             &format!("new/{package}"),
             "-v", // needed to show pass/fail of individual lints
```

_In principle_ cargo should have been able to locate the package and workspace in either case, so this should do nothing. If it changes the outcome, we'll know where to look more closely.

I'm happy to help as best I can, but I can mostly just give suggestions and advice until I can reproduce the issue on my end. I'd appreciate your help in getting this to reproduce reliably!

**Comment by Ashutosh0x** [comment]
Hi @obi1kenobi, I tried reproducing this on Windows with v0.45.0. I'll apply the patch from the comment above and report back with results. If I can get a reliable reproduction, I'd be happy to put together a fix.

**Comment by Ashutosh0x** [comment]
Hey! I just opened PR #1611 adding a union_field_marked_deprecated lint — it mirrors the existing struct_field_marked_deprecated pattern for union fields. I initially tried union_marked_deprecated (PR #1610) but realized it duplicated 	ype_marked_deprecated which already covers unions via ImplOwner. Happy to take on more lints if this one looks good!
